core/src/main/scala/org/apache/gearpump/util/ActorUtil.scala

```diff
@@ -97,4 +99,24 @@ object ActorUtil {
       master.tell(StartExecutorSystems(resources, executorJvmConfig), sender)
     }
   }
- }
+
+
+  def askAppMaster[T](master: ActorRef, appId: Int, msg: Any)(implicit ex: ExecutionContext): Future[T] = {
+    implicit val timeout = Constants.FUTURE_TIMEOUT
+    val appmaster =  askMaster[ResolveAppIdResult](master, ResolveAppId(appId)).flatMap { result =>
+      if (result.appMaster.isSuccess) {
+        Future.successful(result.appMaster.get)
+      } else {
+        Future.failed(result.appMaster.failed.get)
+      }
+    }
+    appmaster.flatMap { appMaster =>
+      (appMaster ? msg).asInstanceOf[Future[T]]
+    }
+  }
+
+  def askMaster[T](master: ActorRef, msg: Any)(implicit ex: ExecutionContext): Future[T] = {
```

Comment:
what does the ```Master``` mean in ```askMaster```? Is it Gearpump Master or AppMaster? Quite confusing to me. To me, ```askMaster``` seems quite like a general ```ask``` wrapper instead of a master communication mechanism.